planner/core/plan.go

```diff
@@ -258,8 +258,10 @@ func (p *basePlan) statsInfo() *property.StatsInfo {
 	return p.stats
 }
 
-func (p *basePlan) ExplainID() string {
-	return fmt.Sprintf("%s_%d", p.tp, p.id)
+func (p *basePlan) ExplainID() func() string {
```

Comment:
Do we have places that called `ExplainID()` but don't execute returned function? e.g.
```go
fn := p.ExplainID()
// fn is not used in the later execution
```

Reply:
fn will be used, normally as executer's field and used by mem tracker, but fn no need be eval until mem tracker trigger action or we run with explain